lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -211,8 +214,15 @@ def add_payment_urls(post, options)
       end
 
       def add_customer_object(post, payment, options)
-        post[:name] = "#{payment.first_name} #{payment.last_name}"
-        post[:phone_number] = options[:billing_address][:phone].gsub(/\D/, '') if options[:billing_address]
+        if options[:customer]
+          post[:name] = options.dig(:customer, :name)
+          post[:phone_number] = options.dig(:customer, :phone_number)
+          post[:email] = options.dig(:customer, :email)
+          post[:addresses] = options.dig(:customer, :addresses)
+        end
+
+        post[:name] = "#{payment.first_name} #{payment.last_name}" unless payment.is_a?(String)
+        post[:phone_number] = options[:billing_address][:phone].gsub(/\D/, '') unless options[:billing_address].nil?
```

Comment:
We should be able to derive these fields from the `billing_address` set of fields, within the `options` hash.

Reply:
We should get the fields from the `options[:billing_address]` instead of `options[:customer]`.